sdk/python/tests/unit/permissions/auth/server/test_auth_registry_server.py

```diff
@@ -65,33 +76,149 @@ def remote_feature_store(server_port, feature_store):
+            ],
+        ).to_df()
+        assertpy.assert_that(training_df).is_not_none()
+
+
+def _test_list_entities(
+    client_fs: FeatureStore, permissions: Union[Permission, list[Permission]]
+):
+    entities = client_fs.list_entities()
+
+    if not _is_auth_enabled(client_fs) or (
+        _is_auth_enabled(client_fs)
+        and (
+            _no_permission_retrieved(permissions)
+            or (
+                _is_listing_permissions_allowed(permissions)
+                and _permissions_exist_in_permission_list(
+                    [list_permissions_perm, list_entities_perm], permissions
+                )
+            )
+        )
+    ):
+        assertpy.assert_that(entities).is_not_none()
+        assertpy.assert_that(len(entities)).is_equal_to(1)
+        assertpy.assert_that(entities[0].name).is_equal_to("driver")
+    else:
+        assertpy.assert_that(entities).is_not_none()
+        assertpy.assert_that(len(entities)).is_equal_to(0)
+
+
+def _no_permission_retrieved(permissions: list[Permission]) -> bool:
+    return len(permissions) == 0
+
+
+def _test_list_permissions(
+    client_fs: FeatureStore, applied_permissions: Union[Permission, list[Permission]]
+) -> list[Permission]:
+    permissions = client_fs.list_permissions()
+
+    if not _is_auth_enabled(client_fs):
+        assertpy.assert_that(permissions).is_not_none()
+        assertpy.assert_that(len(permissions)).is_equal_to(len(applied_permissions))
+    elif _is_auth_enabled(client_fs) and _permissions_exist_in_permission_list(
+        ALL_POSITIVE_TEST_NEEDED_PERMISSIONS, permissions
+    ):
+        assertpy.assert_that(permissions).is_not_none()
+        assertpy.assert_that(len(permissions)).is_equal_to(
+            len(ALL_POSITIVE_TEST_NEEDED_PERMISSIONS)
+        )
+    elif _is_auth_enabled(client_fs) and _is_listing_permissions_allowed(permissions):
+        assertpy.assert_that(permissions).is_not_none()
+        assertpy.assert_that(len(permissions)).is_equal_to(1)
 
-    assertpy.assert_that(entities[0].name).is_equal_to("driver")
+    return permissions
 
 
-def _test_list_fvs(fs: FeatureStore):
-    fvs = fs.list_feature_views()
+def _is_listing_permissions_allowed(permissions: list[Permission]) -> bool:
+    return list_permissions_perm in permissions
+
+
+def _is_auth_enabled(client_fs: FeatureStore) -> bool:
+    return client_fs.config.auth_config.type != "no_auth"
+
+
+def _test_list_fvs(
+    client_fs: FeatureStore, permissions: Union[Permission, list[Permission]]
+):
+    fvs = client_fs.list_feature_views()
     for fv in fvs:
         print(f"{fv.name}, {type(fv).__name__}")
 
-    assertpy.assert_that(fvs).is_not_none()
-    assertpy.assert_that(len(fvs)).is_equal_to(2)
-
-    names = _to_names(fvs)
-    assertpy.assert_that(names).contains("driver_hourly_stats")
-    assertpy.assert_that(names).contains("driver_hourly_stats_fresh")
+    if not _is_auth_enabled(client_fs) or (
```

Comment:
can we move this logic in a function that simplifies the reading, something like:
```py
if _is_permission_enabled(client_fs, permissions, FeatureView, AuthzedAction.READ):
  ...
```